File: Parking management/backend/app/schemas/ocr.py
from pydantic import BaseModel, Field


class PlateOcrResponse(BaseModel):
    success: bool
    plate_number: str | None = None
    raw_text: str = ""
    confidence: float = Field(default=0.0, ge=0.0, le=1.0)
    message: str | None = None

File: Parking management/frontend/src/api/ocr.ts
import { apiRequest } from './client';

export interface PlateOcrResponse {
  success: boolean;
  plate_number: string | null;
  raw_text: string;
  confidence: number;
  message: string | null;
}

export function detectPlate(file: File): Promise<PlateOcrResponse> {
  const formData = new FormData();
  formData.append('file', file);

  return apiRequest<PlateOcrResponse>('/ocr/plate', {
    method: 'POST',
    body: formData,
  });
}
